agent/engine/docker_task_engine_test.go

```diff
@@ -3128,3 +3128,252 @@ func TestStartFirelensContainerRetryForContainerIP(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+	ctrl, client, _, taskEngine, _, _, _ := mocks(t, ctx, &defaultConfig)
+	defer ctrl.Finish()
+	execAgentPID := "1234"
+	resp := &dockercontainer.ContainerTopOKBody{
+		Processes: [][]string{{"root", execAgentPID}},
+	}
+	testTask := &apitask.Task{
+		Arn: "arn:aws:ecs:region:account-id:task/test-task-arn",
+		Containers: []*apicontainer.Container{
+			{
+				Name:      "test-container",
+				RuntimeID: "runtime-ID",
+				ExecCommandAgentMetadata: &apicontainer.ExecCommandAgentMetadata{
+					PID: execAgentPID,
+				},
+			},
+		},
+		ExecCommandAgentEnabled: true,
+	}
+	taskEngine.(*DockerTaskEngine).state.AddTask(testTask)
+	taskEngine.(*DockerTaskEngine).managedTasks[testTask.Arn] = &managedTask{Task: testTask}
+	client.EXPECT().TopContainer(gomock.Any(), testTask.Containers[0].RuntimeID, gomock.Any(), execAgentPID).Return(resp, nil)
+	taskEngine.(*DockerTaskEngine).monitorExecAgentProcesses(ctx)
+	time.Sleep(5 * time.Millisecond)
+	assert.Equal(t, execAgentPID, testTask.Containers[0].GetExecCommandAgentMetadata().PID)
+}
+
+func TestMonitorExecAgentProcessExecDisabled(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+	ctrl, _, _, taskEngine, _, _, _ := mocks(t, ctx, &defaultConfig)
+	defer ctrl.Finish()
+	testTask := &apitask.Task{
+		Arn: "arn:aws:ecs:region:account-id:task/test-task-arn",
+		Containers: []*apicontainer.Container{
+			{
+				Name:      "test-container",
+				RuntimeID: "runtime-ID",
+			},
+		},
+	}
+	taskEngine.(*DockerTaskEngine).state.AddTask(testTask)
+	taskEngine.(*DockerTaskEngine).managedTasks[testTask.Arn] = &managedTask{Task: testTask}
+	taskEngine.(*DockerTaskEngine).monitorExecAgentProcesses(ctx)
+	time.Sleep(5 * time.Millisecond)
+}
+func TestMonitorExecAgentsMultipleContainers(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+	ctrl, client, _, taskEngine, _, _, _ := mocks(t, ctx, &defaultConfig)
+	defer ctrl.Finish()
+	execAgentPID1 := "1234"
+	execAgentPID2 := "2345"
+	resp := &dockercontainer.ContainerTopOKBody{
+		Processes: [][]string{{"root", execAgentPID1}},
+	}
+	testTask := &apitask.Task{
+		Arn: "arn:aws:ecs:region:account-id:task/test-task-arn",
+		Containers: []*apicontainer.Container{
+			{
+				Name:      "test-container1",
+				RuntimeID: "runtime-ID1",
+				ExecCommandAgentMetadata: &apicontainer.ExecCommandAgentMetadata{
+					PID: execAgentPID1,
+				},
+			},
+			{
+				Name:      "test-container2",
+				RuntimeID: "runtime-ID2",
+				ExecCommandAgentMetadata: &apicontainer.ExecCommandAgentMetadata{
+					PID: execAgentPID2,
+				},
+			},
+		},
+		ExecCommandAgentEnabled: true,
+	}
+	taskEngine.(*DockerTaskEngine).state.AddTask(testTask)
+	taskEngine.(*DockerTaskEngine).managedTasks[testTask.Arn] = &managedTask{Task: testTask}
+	client.EXPECT().TopContainer(gomock.Any(), testTask.Containers[0].RuntimeID, gomock.Any(), execAgentPID1).Return(resp, nil)
```

Comment:
I think we should also include the timeout in the EXPECT call, so that if we change that value either intentionally or accidentally in the future, we have a test that makes us aware of the change. (let's not use the constant but directly the expected value)

Reply:
I'm not quite sure I understand the comment. Could you give an example?

Reply:
just like 

```golang
client.EXPECT().TopContainer(gomock.Any(), testTask.Containers[0].RuntimeID, time.Minute * 15, execAgentPID1).Return(resp, nil)
```